app.js

```diff
@@ -103,18 +104,15 @@ var processRequest = function (req, res, isTar) {
     requests[id]['version'] = meta.version;
     requests[id]['version-specberus'] = SpecberusWrapper.version;
     requests[id]['decision'] = decision;
-    var jobList;
+    var jobList = ['retrieve-resources', 'metadata', 'specberus', 'third-party-checker', 'publish', 'tr-install', 'update-tr-shortlink'];
 
-    if (isTar) {
-      jobList = ['retrieve-resources', 'metadata', 'user-checker', 'specberus',
-                 'third-party-checker', 'publish', 'tr-install',
-                 'update-tr-shortlink'];
-    }
-    else {
-      jobList = ['retrieve-resources', 'metadata', 'specberus', 'token-checker',
-                 'third-party-checker', 'publish', 'tr-install',
-                 'update-tr-shortlink'];
-    }
+    if (isTar)
+      jobList.splice(2, 0, 'user-checker');
+    else
+      jobList.splice(3, 0, 'token-checker');
+
+    if (dryRun)
+      jobList.splice(jobList.indexOf('publish'));
```

Comment:
@deniak, it does that ([`Array.prototype.splice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice#Parameters)).